src/jl_uv.c

```diff
@@ -678,56 +678,71 @@ JL_DLLEXPORT int jl_printf(uv_stream_t *s, const char *format, ...)
     return c;
 }
 
-STATIC_INLINE void print_error_msg_as_json(char *buf) JL_NOTSAFEPOINT
+STATIC_INLINE int copystp(char *dest, const char *src)
 {
-    // Our telemetry on SPCS expects a JSON object per line
-    // The following lines prepare the timestamp string and the JSON object
+    char *d = stpcpy(dest, src);
+    return (int)(d - dest);
+}
+
+// RAI-specific
+STATIC_INLINE void write_to_safe_crash_log(char *buf) JL_NOTSAFEPOINT
+{
+    // Our telemetry on SPCS expects a JSON object per line.
+    // We ignore write failures because there is nothing we can do.
+    int buflen = strlen(buf);
+    assert(buflen <= 4096);
+    char wbuf[4096];
```

Comment:
Why 4096? Why not strlen + the Json overhead?

Reply:
Also, on reflection, if there's some maximum size, could you print it in multiple messages?

Reply:
This is called by `jl_safe_printf()`. As far as I can see, it is not being called with any large text, which is partly why I chose this approach. My first attempt was `alloca`, but that was problematic. I'm trying to avoid `malloc` which will add a lot of overhead, but that's the only way I can see to avoid a limit like 4096 and the `assert`.

Reply:
I chose your first suggestion -- to truncate if the message is too large -- mainly because I'm pretty sure that it will almost never happen because of the way `jl_safe_printf` is used.